eclipse-theia/theia
# plugin: add workspace file api

#### What it does

Add `on(will|did)(create|rename|delete)Files` to the plugin's workspace
api. This implementation does not handle the `WorkspaceEdit` apis.

Fixes: https://github.com/eclipse-theia/theia/issues/7171

#### How to test

1. include the following [plugin](https://github.com/vince-fugnitto/workspace-event-api/releases/download/v1.0.0/workspace-api-ext-0.0.1.vsix)
2. perform **add / rename / delete** of files through the user-interface and and check that events are correctly fired.

#### Review checklist


#### Reminder for reviewers

- as a reviewer, I agree to behave in accordance with [the review guidelines](https://github.com/theia-ide/theia/blob/master/doc/pull-requests.md#reviewing)
